apps/home/blocks.py

```diff
@@ -0,0 +1,29 @@
+from django.utils.translation import gettext as _
+from wagtail.core import blocks
+
+
+class SectionBlock(blocks.StructBlock):
+    section = blocks.ChoiceBlock(
+        choices=[
+            ("tutorial", _("Tutorial")),
+            ("how-to", _("How-to")),
+            ("reference", _("Reference")),
+            ("explanation", _("Explanation")),
+        ],
+        label=_("Section"),
+    )
+    title = blocks.CharBlock(label=_("Title"))
+    text = blocks.TextBlock(label=_("Text"))
+
+    class Meta:
+        template = "home/section_block.html"
+
+
+class SectionGridBlock(blocks.ListBlock):
+    def __init__(self, *args, **kwargs):
+        kwargs.update({"child_block": SectionBlock(label=_("Section"))})
+        super().__init__(*args, **kwargs)
+
```

Comment:
Can we add a get context here?
```
    def get_context(self, value, parent_context=None):
        context = super().get_context(value, parent_context)
        context["icon_template"] = f"/path/to/{self.bla}.svg"
        return context
        ```
        
to replace `apps/home/templates/home/section_block.html`  line 4.

Reply:
I guess we cannot access class attributes like `self.section`, `self.title` etc. inside methods in `StructBlock`.  Here is a issue I found with same problem https://github.com/wagtail/wagtail/issues/4387. We can extend `StructValue` and define a method there. I don't know if we can define `get_context` method this way.